src/jdk.jpackage/windows/native/libwixhelper/libwixhelper.cpp

```diff
@@ -72,10 +72,10 @@ class ProductInfo {
 
 void findInstalledProducts(const Guid& upgradeCode,
                                             std::vector<ProductInfo>& products) {
-    const LPCTSTR upgradeCodeStr = upgradeCode.toMsiString().c_str();
+    const tstring upgradeCodeStr = upgradeCode.toMsiString();
```

Comment:
I am not familiar with this warning. Does it mean that in original `const LPCTSTR upgradeCodeStr = upgradeCode.toMsiString().c_str();`, `tstring` retuned from `toMsiString()` will get destroyed right after` c_str()` completed? Which will make `upgradeCodeStr` invalid?

Reply:
Yep, that is correct